### ISSUE TITLE
Google DNS not working - potentially need to set user agent string

### ISSUE THREAD

**Issue by jfritz** [TimothyYe/godns#46]
godns 1.9.1 doesn't seem to successfully update my ddns with Google. The messaging looks successful:

```
C:\Users\jeff\Desktop\GoDNS>godns.exe
[GoDNS] 2019/08/30 18:46:20 GoDNS started, entering main loop...
[GoDNS] 2019/08/30 18:46:20 Creating DNS handler with provider: Google
[GoDNS] 2019/08/30 18:46:21 currentIP is: [censored]
[GoDNS] 2019/08/30 18:46:21 Going to sleep, will start next checking in 300 seconds...
```

however, checking Google Domains, the last update for my DNS was stated to be months ago.

To check, I made the request directly via a browser to the end point using the same config/username/password/domain, and this successfully updated my ddns IP.
`https://CENSORED:CENSORED@domains.google.com/nic/update?hostname=CENSORED.net&myip=CENSORED`

I noticed that `UpdateIP()` in the Google handler does not set a user agent when sending the request to the Google API endpoint. Per Google's API [documentation](https://support.google.com/domains/answer/6147083?hl=en), you should set a user agent when updating DNS: "Note: You must set a user agent in your request as well. Web browsers will generally add this for you when testing via the above url."

Note the user agent might be required for other services as well, as it is one of the few things public API endpoints can use to ban "bad" bots who improperly use APIs without actually banning specific IPs directly - not saying this is the case here, but just noting in general.

Edit: is there a way I can set user agent via the configuration?

**Comment by TimothyYe** [comment]
Thanks for your feedback, I'll add it.

**Comment by TimothyYe** [comment]
@jfritz Please try this version https://github.com/TimothyYe/godns/releases/tag/V1.9.2
You can set the user agent via the configuration now.

**Comment by jfritz** [comment]
Alright, the user agent was probably an issue, but I dug deeper (since it still wasn't working) and found another problem (which I hope is probably the last/root cause issue). Sorry for all the hassle here.

The `username:password` need to be provided via HTTP Basic Auth, not as query string parameters or POST data as the code is currently doing. Below is an example from the [Google Domains API documentation](https://support.google.com/domains/answer/6147083?hl=en). 

Tl;dr - this might be as simple as using `req.SetBasicAuth(username, password)` and removing the `values.Add("username:password", ...)`. 

![2019-09-03 20_56_13-Window](https://user-images.githubusercontent.com/139331/64220033-85157f80-ce8d-11e9-9cfc-8f6535953041.png)

**Comment by TimothyYe** [comment]
Really appreciate with your feedback.
I don't have a google domain account since it is not supported in my country.
You may try this version: https://github.com/TimothyYe/godns/releases/tag/V1.9.3

**Comment by TimothyYe** [comment]
Hi @jfritz , is this issue fixed already? I plan to close this issue...

**Comment by hemming1** [comment]
Hi @TimothyYe, have you confirmed with any test users that Google DNS Update is working?

I'm getting the following feedback in a Windows environment:

`E:\Repositories\godns-2.1>godns.exe
[GoDNS] 2020/04/22 16:09:23 GoDNS started, entering main loop...
[GoDNS] 2020/04/22 16:09:23 Creating DNS handler with provider: Google
[GoDNS] 2020/04/22 16:09:24 currentIP is: [censored]
[GoDNS] 2020/04/22 16:09:24 [censored] Start to update record IP...
[GoDNS] 2020/04/22 16:09:24 Update IP success: notfqdn
[GoDNS] 2020/04/22 16:09:24 Going to sleep, will start next checking in 21600 seconds...
`

However, updating manually in the browser works 'https://CENSORED:CENSORED@domains.google.com/nic/update?hostname=CENSORED.net&myip=CENSORED'

**Comment by TimothyYe** [comment]
@hemming1 Hmmm, Google Domain is unavailable in my country, however, I'll find a way to register an account and try to reproduce this issue.

**Comment by hemming1** [comment]
@jfritz Were you able to get your domain ip to update successfully?

**Comment by jfritz** [comment]
I ultimately I ran into crunch time at work and it prevented me from
debugging the issue further. I did get the direct web request updates to
work as you mentioned earlier.

**Comment by BradHemmingsen** [comment]
@TimothyYe  would it be useful if I setup a temporary domain for you to test on?

**Comment by TimothyYe** [comment]
@BradHemmingsen That would be great! Thanks!

**Comment by TimothyYe** [comment]
@BradHemmingsen Now I get another error, it seems that DDNS is not enabled for this temporary domain. Would you pls help to enable it? 

```
nohost - The hostname does not exist, or does not have Dynamic DNS enabled.
```

**Comment by BradHemmingsen** [comment]
When I registered the domain this morning it said it might take 24-48
hours. I'll give you credentials to one of my real ones.

wrt1900ac.bradhemmingsen.com

*Username:* KvuWPEOqEu6iResU
*Password:* wbfssPagJ9IIFmwA

**Comment by TimothyYe** [comment]
@BradHemmingsen Issue has been fixed, pls try this pre-release version:
https://github.com/TimothyYe/godns/releases/tag/V2.1.1
